src/borg/remote.py

```diff
@@ -237,12 +243,27 @@ def borg_cmd(self, args, testing):
     def ssh_cmd(self, location):
         """return a ssh command line that can be prefixed to a borg command line"""
         args = shlex.split(os.environ.get('BORG_RSH', 'ssh'))
-        if location.port:
-            args += ['-p', str(location.port)]
-        if location.user:
-            args.append('%s@%s' % (location.user, location.host))
+        if sys.platform == 'win32' and 'BORG_RSH' not in os.environ:
+            if shutil.which('ssh') is not None:
+                args = ['ssh']
+            elif shutil.which('plink') is not None:
+                args = ['plink', '-ssh', '-batch']
+            else:
+                raise Exception('Could not find supported ssh client.')
+        if args[0] == 'plink':
+            if location.port:
+                args += ['-P', str(location.port)]
+            if location.user:
+                args.append('%s@%s' % (location.user, location.host))
+            else:
+                args.append('%s' % location.host)
         else:
-            args.append('%s' % location.host)
+            if location.port:
+                args += ['-p', str(location.port)]
+            if location.user:
+                args.append('%s@%s' % (location.user, location.host))
+            else:
+                args.append('%s' % location.host)
```

Comment:
most of the if/else block is the same. are we expecting to have to support more differences than just -P vs -p?

if not, it could be unified and just use a "'-P' if plink else '-p'" expression.